web/packages/teleport/src/Navigation/SideNavigation/Section.tsx

```diff
@@ -0,0 +1,256 @@
+import React from 'react';
+import { NavLink } from 'react-router-dom';
+import styled, { css } from 'styled-components';
+
+import { Box } from 'design';
+import * as Icons from 'design/Icon';
+import { Theme } from 'design/theme';
+
+import { CustomNavigationCategory, NavigationCategory } from './categories';
+import { NavigationSection } from './Navigation';
+import { zIndexMap } from './zIndexMap';
+
+export function Section({
+  section,
+  $active,
+  setExpandedSection,
+  onClick,
+  children,
+  isExpanded,
+}: {
+  section: NavigationSection;
+  $active: boolean;
+  setExpandedSection: () => void;
+  onClick: (event: React.MouseEvent) => void;
+  isExpanded?: boolean;
+  children?: JSX.Element;
+}) {
+  return (
+    <>
+      <CategoryButton
+        $active={$active}
+        onMouseEnter={setExpandedSection}
+        onFocus={setExpandedSection}
+        onClick={onClick}
+        isExpanded={isExpanded}
+        tabIndex={section.standalone ? 0 : -1}
+      >
+        <CategoryIcon category={section.category} />
+        {section.category}
+      </CategoryButton>
+      {children}
+    </>
+  );
+}
+
+const rightPanelWidth = '236px';
+
+export const RightPanel = styled(Box).attrs({ pt: 2, px: 2 })<{
+  isVisible: boolean;
+  skipAnimation: boolean;
+}>`
+  position: fixed;
+  left: var(--sidenav-width);
+  height: 100%;
+  scrollbar-gutter: auto;
+  overflow: visible;
+  width: ${rightPanelWidth};
+  background: ${p => p.theme.colors.levels.surface};
+  z-index: ${zIndexMap.sideNavExpandedPanel};
+  border-right: 1px solid ${p => p.theme.colors.spotBackground[1]};
+
+  ${props =>
+    props.isVisible
+      ? `
+      ${props.skipAnimation ? '' : 'transition: transform .15s ease-out;'}
+      transform: translateX(0);
+      `
+      : `
+      ${props.skipAnimation ? '' : 'transition: transform .15s ease-in;'}
+      transform: translateX(-100%);
+      `}
+
+  top: ${p => p.theme.topBarHeight[0]}px;
+  padding-bottom: ${p => p.theme.topBarHeight[0] + p.theme.space[2]}px;
+  @media screen and (min-width: ${p => p.theme.breakpoints.small}px) {
+    top: ${p => p.theme.topBarHeight[1]}px;
+    padding-bottom: ${p => p.theme.topBarHeight[1] + p.theme.space[2]}px;
+  }
+  @media screen and (min-width: ${p => p.theme.breakpoints.large}px) {
+    top: ${p => p.theme.topBarHeight[2]}px;
+    padding-bottom: ${p => p.theme.topBarHeight[3] + p.theme.space[2]}px;
+  }
+`;
+
+export function CategoryIcon({
```

Comment:
`CategoryIcon` function is defined in the `CategoryIcon.tsx` file as well? Can we use a single shared function for navigation category?